src/rootUtils.h

```diff
@@ -272,6 +272,30 @@ inline std::vector<std::string> sortAlphabeticaly(std::vector<std::string> strin
   return strings;
 }
 
+/**
+ * Check whether existingColls and candidateColls both contain the same
+ * collection names. Returns false if the two vectors differ in content. Inputs
+ * can have random order wrt each other, but the assumption is that each vector
+ * only contains unique names.
+ */
+inline bool checkConsistentColls(const std::vector<std::string>& existingColls,
+                                 const std::vector<std::string>& candidateColls) {
```

Comment:
We should print out which collections are missing, and not return on the first one we don't find.

Reply:
Should we make that part of the exception message? (That might become quite long then). An alternative approach would be to keep the exception in its terse form and then give access to the `collsToWrite` for each category so that analysis can be done on the user side, resp. we could provide something like
```cpp
using namespace std;
tuple<vector<string>, vector<string>> checkConsistency(const vector<string>& collsToWrite, const string& category);
```
which returns the collection names that only exist in either the passed `collsToWrite` or inside the `category`. This could in principle then also be used as a pre-check, before calling `writeFrame`.

Reply:
Making it part of the exception is also good. But we must log it somewhere! Rather a too long exception message than a too short one.

Reply:
This is now part of the exception message and there is the possibility to check for consistency.